erpnext/accounts/doctype/pos_invoice/pos_invoice.py

```diff
@@ -212,6 +213,25 @@ def validate_serialised_or_batched_item(self):
 		if error_msg:
 			frappe.throw(error_msg, title=_("Invalid Item"), as_list=True)
 
+	def validate_serial_nos(self):
+		error_msg = []
+		for item in self.get("items"):
+			serialized = item.get("has_serial_no")
+			msg = ""
+			if serialized:
+				valid_serial_nos = frappe.get_all('Serial No', filters={'item_code':item.get("item_code")}, pluck="name")
```

Comment:
maybe only active ones should be fetched?

Reply:
This still means same serial no can be sold twice if not reserved like for batches 😅